gensim/models/ldamodel.py

```diff
@@ -1054,13 +1059,22 @@ def load(cls, fname, *args, **kwargs):
             result.state = super(LdaModel, cls).load(state_fname, *args, **kwargs)
         except Exception as e:
             logging.warning("failed to load state from %s: %s", state_fname, e)
-        id2word_fname = utils.smart_extension(fname, '.id2word')
-        if (os.path.isfile(id2word_fname)):
-            try:
-                result.id2word = utils.unpickle(id2word_fname)
-            except Exception as e:
-                logging.warning("failed to load id2word dictionary from %s: %s", id2word_fname, e)
+
+        random_state_fname = utils.smart_extension(fname, '.random_state')
+        if (os.path.isfile(random_state_fname)):
+            result.random_state = utils.unpickle(random_state_fname)
         else:
-            result.id2word = None
+            logging.warning("random_state not stored on disk so using default value")
```

Comment:
@tmylk Your suggestion is a bit different from the [earlier discussion](https://github.com/RaRe-Technologies/gensim/issues/1082#issuecomment-301164932) on the issue. Hence, I want to make sure I understand the desired solution before making the changes.

If I understand it correctly, what you are suggesting is :  
- If there is indeed a file with the extension `.random_state` present on disk, this means that the model was saved using a pre-0.13.2 version of Gensim. So we use this file to set `result.random_state` at the time of loading. 
- However, if there is no such file present on disk, then this means that the model was saved using a post-0.13.2 version of Gensim and thus `result.random_state` got set at the time of the main pickle load. So in this case, we don't need to do anything else.

But for models saved using a pre-0.13.2 version of Gensim, there was no `.random_state` file created at the time of saving the model. So while loading such a model from disk, where would the `.random_state` file come from in this case? Is the user responsible for creating this file explicitly in such a case? If this is true, then I believe we don't need to make any changes in the `save` function for LdaModel at all and we just need to change the `load` function. 

Please correct me if I am wrong or missing something here. Otherwise, if this is indeed what we need, I could go ahead and make the appropriate changes.